src/components/subscribemodal/index.tsx

```diff
@@ -15,11 +14,15 @@ const SubscribedModal: React.FC<SubscribedModalProps> = ({ onClose }) => {
   };
 
   // Form signer and contract connection
-  ReactUse.useClickAway(modalRef, onClose);
+    window.addEventListener("click", (e: MouseEvent) => {
```

Comment:
this is an imperative way of doing things, we should create hooks and reduce the code present in the usage. also don't use ID. Our hook should have a flexible API atleast like `react-use`

```
useClickAway(ref, onMouseEvent)
useClickAway(ref, onMouseEvent, ['click'])
useClickAway(ref, onMouseEvent, ['mousedown', 'touchstart'])
```

You can actually refer react-use itself to see how they are doing.

Reply:
Can we can import useClickAway only from react-use and use this line useClickAway(modalRef, onClose) directly? 
We are not using anything else from react-use.